test/box-tap/gh-6647-yield-gc-hook.test.lua

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env tarantool
+
+local tap = require('tap')
+local test = tap.test('gh-6647-yield-gc-hook-iproto-request')
+local fio = require('fio')
+test:plan(1)
+
+local tarantool_bin = arg[-1]
+local PANIC = 256
+
+local function run_script(code)
+    local dir = fio.tempdir()
+    local script_path = fio.pathjoin(dir, 'script.lua')
+    local script = fio.open(script_path, {'O_CREAT', 'O_WRONLY', 'O_APPEND'},
+        tonumber('0777', 8))
+    script:write(code)
+    script:write("\nos.exit(0)")
+    script:close()
+    local cmd = [[/bin/sh -c 'cd "%s" && "%s" ./script.lua 2> /dev/null']]
+    local res = os.execute(string.format(cmd, dir, tarantool_bin))
+    fio.rmtree(dir)
+    return res
+end
+
+local code = [[
+box.cfg({listen = 3301})
+box.schema.user.grant('guest', 'super')
+ffi = require('ffi')
+fiber = require('fiber')
+net_box = require('net.box')
+c = net_box.connect(3301)
+local function f()
+    local t = {__gh_hook = ffi.gc(ffi.new('void *'), fiber.yield)}
+    t = nil
+    collectgarbage()
+end
+c:call('f')
+os.exit(0)
```

Comment:
`run_script()` adds `os.exit()` on its own. It is harmless, but I would do it once.

I would leave `os.exit()` in the script itself, because this way no parts of the script will spread across testing code. However it is just my feelings, feel free to ignore if you feels it differently.